### PR TITLE
fix: the --base option not work

### PR DESCRIPTION
**Pull request by byyuurin**
### Description

I noticed that vite-ssg build provides the `--base` option, but I found that it doesn't work as expected when I tried to use it.

Here are the tests I conducted after fixing this issue:

- Set `base` in vite.config.ts and perform a build to confirm the result.
- Set `ssgOptions.base` in vite.config.ts and perform a build to confirm the result.
- Add `--base` to vite-ssg build command and perform a build to confirm the result.
- Add `--mode` to vite-ssg build command and perform a build to confirm the result.
- Add both `--mode` and `--base` to vite-ssg build command and perform a build to confirm the result.
- Add `NODE_ENV` and `--mode` to vite-ssg build command and perform a build to confirm the result.

### Linked Issues

#339 #340 #341

### CODE REVIEW

**Comment by antfu** [comment]
Can you help resolve the conflicts? Thanks

**Comment by byyuurin** [comment]
Ok, updated.

File: src/node/build.ts [hunk 1]
```diff
@@ -27,8 +27,9 @@ function DefaultIncludedRoutes(paths: string[], _routes: Readonly<RouteRecordRaw
 }
 
 export async function build(ssgOptions: Partial<ViteSSGOptions> = {}, viteConfig: InlineConfig = {}) {
-  const mode = process.env.MODE || process.env.NODE_ENV || ssgOptions.mode || 'production'
-  const config = await resolveConfig(viteConfig, 'build', mode, mode !== 'development' ? 'production' : 'development')
+  const nodeEnv = process.env.NODE_ENV || 'production'
+  const mode = process.env.MODE || ssgOptions.mode || nodeEnv
+  const config = await resolveConfig(viteConfig, 'build', mode, nodeEnv)
```
Comment (antfu) [inline src/node/build.ts:32]: This seems to miss the fix in https://github.com/antfu/vite-ssg/pull/362

Comment (byyuurin) [reply src/node/build.ts:32]: After reviewing the fix in #362, I noticed that this modification has compromised the flexibility of adjusting NODE_ENV.

For example:

```bash
nr cross-env NODE_ENV=development vite-ssg build --mode ssg
```

I believe that the logic for NODE_ENV and MODE should not be merged together.

**Pull request merged**